src/main/java/com/aws/greengrass/util/CommitableFile.java

```diff
@@ -117,14 +117,15 @@ public void abandon() {
     }
 
     /**
-     * Close the file and commit the new version.  The old version becomes a backup
+     * Close the file and commit the new version. The old version becomes a backup
      */
     @SuppressWarnings("ConvertToTryWithResources")
     @Override
     public void commit() {
         if (!closed) {
             try {
-                super.close();
+                flush();
```

Comment:
is there a difference between flush() and out.flush() ?

Reply:
No, flush() will redirect to out.flush(). But this way if we add more stuff into flush for some reason, it would get called here.